language/e2e-tests/src/data_store.rs

```diff
@@ -113,8 +119,51 @@ impl StateView for FakeDataStore {
 }
 
 // This is used by the `process_transaction` API.
-impl RemoteCache for FakeDataStore {
+/*impl RemoteCache for FakeDataStore {
     fn get(&self, access_path: &AccessPath) -> VMResult<Option<Vec<u8>>> {
         Ok(StateView::get(self, access_path).expect("it should not error"))
     }
+}*/
+
+impl RemoteCache for FakeDataStore {
```

Comment:
Yes, the current implementation is a bit messy. Alternatively, maybe instead of implementing `RemoteCache` for `FakeDataStore` directly, we should just wrap it inside a `RemoteStorage`.

Anyways it's a bit hard for me to tell exactly what the details should be right now, as I'm still finalizing the `RemoteCache` API and may take a step back, making `get_module` and `get_resource` return bytes instead. I'll review the whole story later.